flow/connectors/postgres/client.go

```diff
@@ -588,22 +588,26 @@ func (c *PostgresConnector) CheckSourceTables(tableNames []string, pubName strin
 	}
 
 	// Check that we can select from all tables
-	for _, tableName := range tableNames {
+	tableArr := make([]string, 0, len(tableNames))
+	for _, table := range tableNames {
 		var row pgx.Row
-		err := c.pool.QueryRow(c.ctx, fmt.Sprintf("SELECT * FROM %s LIMIT 0;", tableName)).Scan(&row)
+		schemaName, tableName, found := strings.Cut(table, ".")
+		if !found {
+			return fmt.Errorf("invalid source table identifier: %s", table)
+		}
+
+		quotedTableIdentifier := fmt.Sprintf("%s.%s",
+			QuoteLiteral(schemaName),
+			QuoteLiteral(tableName))
+		tableArr = append(tableArr, quotedTableIdentifier)
```

Comment:
This should be unescaped given SQL its used in later, `fmt.Sprintf("%s.%s", schemaName, tableName)` but ideally there wouldn't be string building in sql, maybe if you built this as a CTE for query
```sql
with checknames_comeupwithabettername (sname, tname) as ( values (
  %s
))
```
Where %s is replaced by `strings.Join(tableArr, ",")`
```go
tableArr := make([]string, 0, len(tableNames))
for _, table in tableNames {
  ...
  tableArr = append(tableArr, fmt.Sprintf("(%s, %s)", QuoteLiteral(schemaName), QuoteLiteral(tableName)))
}
```
then in query join the CTE with `pg_publication_tables`